### ISSUE TITLE
Adopting GSA's Error Identification

### ISSUE THREAD

**Issue by angelina-smith** [GSA/px-benefit-finder#1121]
### Feedback from GSA accessibility audit team
See [success criteria 3.3.3](https://www.w3.org/WAI/WCAG21/Understanding/error-suggestion.html). The error message should say what was not filled out and link them to that form section to complete the form.
Reference: USAGov's error handling at: https://www.usa.gov/elected-officials#input-street.

![image](https://github.com/GSA/px-benefit-finder/assets/142342010/7501e871-cc9e-48f0-94f8-71457d9a7268)

### What happened

- On the About the application page using NVDA the user does not select a Marital Status the error message only says that all fields are required but does not say what was missed.

From AT UT sessions:
- When user makes an error in the Year field alone, the Month and Day fields are also flagged as being invalid.

### How to replicate

Create an error by leaving required fields empty or making a mistake in memorable date.

**Comment by angelina-smith** [comment]
@fongcindy I know this touches on multiple practice areas and will require more conversation. However, this is something the GSA a11y team and our accessibility team should be implement as it addresses multiple accessibility failures. (I also feel like all of the feedback from our AT UT sessions supports the need for our improved error identification/handling)

**Comment by ErnestoFernandezAlvarez** [comment]
@Cindy Fong - QQBC-C ***@***.***> @Angelina Smith - QQBC-C
***@***.***> I wanted to mention that USA.gov has a preferred
way of doing error messages. We briefly addressed Error Messages during
their design feedback sessions. I think when we get to this stage we should
regroup and look into this in detail (devs, usa.gov, a11y, design). Coming
off the AT sessions, I thought this feature was the most problematic.

**Comment by angelina-smith** [comment]
History and context for where this ticket comes from: Both the feedback from all AT UT sessions and USAgov's accessibility audit in early Jan shows that we need to adopt USAgov's error identification.

**Comment by fongcindy** [comment]
@ErnestoFernandezAlvarez No prototype or design is needed for this for now, Angelina's screenshot above is enough.

Next step: 
Scott to see how far he can fix on his end without design for now, will keep us posted.

**Comment by ErnestoFernandezAlvarez** [comment]
Hi team
@scottqueen-bixal @fongcindy @angelina-smith @erniedeeb 

Please note that there isn't design's input on this ticket at the moment. Cindy confirmed Scott will be handing this ticket. If there is a need from design I will be notified. 
Thank you!

**Comment by ErnestoFernandezAlvarez** [comment]
Sounds good, thanks!

**Comment by scottqueen-bixal** [comment]
I've broken this up into three workable issues

[Include additional text to resolve form field errors with fields](https://github.com/GSA/px-benefit-finder/issues/1513)	https://github.com/GSA/px-benefit-finder/issues/1513
[Include a list of links to the fields that have errors in alerts](https://github.com/GSA/px-benefit-finder/issues/1512)	https://github.com/GSA/px-benefit-finder/issues/1512
[Include error count in alert](https://github.com/GSA/px-benefit-finder/issues/1511)	https://github.com/GSA/px-benefit-finder/issues/1511

**Comment by scottqueen-bixal** [comment]
This work was pushed to sprint 37 to prioritize work on the dataLayer
